lib/nrf_modem_lib/Kconfig.modemlib

```diff
@@ -220,10 +220,10 @@ config NRF_MODEM_LIB_ON_FAULT_APPLICATION_SPECIFIC
 	  Let the application define the fault handler function.
 
 config NRF_MODEM_LIB_ON_FAULT_LTE_NET_IF
-	bool "Application defined"
+	bool "Pass to network interface driver"
 	depends on NRF_MODEM_LIB_NET_IF
 	help
-	  Let the application define the fault handler function.
+	  Pass modem faults to the network interface driver.
```

Comment:
We should mention it's a `NET_EVENT_CONN_IF_FATAL_ERROR` error in the help text, so the user knows which event to handle.